Use the common frequenz API

tiyash-basu-frequenz opened the pull request.

tiyash-basu-frequenz committed:
Add submodule frequenz-api-common v0.2.0

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz committed:
Use proto definitions from the common API

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

leandro-lucarella-frequenz reviewed (commented):
LGTM. I would add a release notes entry mentioning the split, as downstream projects could be accessing some of the common objects or enums, and now they need to be imported from a different place in python, and if used explicitly, the `frequenz-api-common` pip package should be added to the dependencies too.

tiyash-basu-frequenz committed:
Use common proto API's pypi package in python lib

Python files for proto definitions in `frequenz-api-common` are no longer
being generated using protoc. Instead, the pypi package
`frequenz-api-common` is being added as a dependency to the python package
definition.
This is required, otherwise each proto library depending on
`frequenz-api-common` will generate its own python modules for
`frequenz-api-common`, resulting in multiple definition of the same common
structures.

Note that this forces the minimum required python version to be 3.11, as a
transitive dependency from the `frequenz-api-common` package.

Signed-off-by: Tiyash Basu <tiyash.basu@frequenz.com>

tiyash-basu-frequenz commented:
> I would add a release notes entry mentioning the split, as downstream projects could be accessing some of the common objects or enums, and now they need to be imported from a different place in python, and if used explicitly, the frequenz-api-common pip package should be added to the dependencies too.

Of course.
This is still a draft PR, and is WIP. I'll update people when it's ready for a review.

tiyash-basu-frequenz commented:
Closed this because the tests were waiting indefinitely for the outdated python versions.
I will create a new PR for this instead.
